components/eventing-controller/pkg/backend/eventmesh/eventmesh.go

```diff
@@ -0,0 +1,314 @@
+package eventmesh
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"go.uber.org/zap"
+
+	apigatewayv1beta1 "github.com/kyma-incubator/api-gateway/api/v1beta1"
+
+	eventingv1alpha2 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2"
+	"github.com/kyma-project/kyma/components/eventing-controller/logger"
+	backendbebv1 "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/beb"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/cleaner"
+	backendutils "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/utils"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/ems/api/events/client"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/ems/api/events/types"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/ems/auth"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/ems/httpclient"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
+)
+
+const (
+	eventMeshHandlerName               = "event-mesh-handler"
+	MaxEventMeshSubscriptionNameLength = 50
+	EventMeshTypeSegmentsLimit         = 7
```

Comment:
```suggestion
	eventTypeSegmentsLimit         = 7
```

Reply:
its used outside the pkg as well

Reply:
are you sure? 
Also needed to remove/rename the package name from the start

Reply:
updated